File: flutter17_mobile/lib/models/role.dart
import 'package:json_annotation/json_annotation.dart';

part 'role.g.dart';

@JsonSerializable()
class Role {
  int? id;
  String? roleName;
  bool? isDeleted;

  Role({
    this.id,
    this.roleName,
    this.isDeleted,
  });

  factory Role.fromJson(Map<String, dynamic> json) => _$RoleFromJson(json);

  Map<String, dynamic> toJson() => _$RoleToJson(this);
}

File: angular17_web/src/app/models/role/role.ts
export class Role {
  id!: number;
  roleName!: string;
  isDeleted!: boolean;

  constructor(data?: any) {
    if (!data) return;

    this.id = data.id;
    this.roleName = data.roleName;
    this.isDeleted = data.isDeleted;
  }
}
